Automattic/simplenote-ios
# Wiring NotesListController

### Details:
In this PR we're replacing **NoteListViewController's** NSFetchedResultsController with the shiny and new **NotesListController**.

This will allow us to retrieve matching Tags and Notes, (and down the line History entries), while maintaining a lightweight ViewController.

**Note:** this PR is all about logic. UI adjustments come in follow up PR's!

@bummytime Ready for review sir!! thank youuuu!!
Ref. #507

### Test: Fresh Install
1. Fresh install the app
2. Log into your account

- [x] Verify your Notes immediately show up
- [x] Verify that navigating thru the Tags do filter out the relevant notes
- [x] Verify that the Title matches the selected Tag
- [x] Verify that the App's SortMode is properly applied
- [x] Verify that Swiping over any of the Notes reveal the Swipeable Actions

### Test: Search + Notes + 3d Touch
1. Log into your account
2. Perform 3d Touch over any of your notes (undeleted)

- [x] Verify that you can push the editor without trouble

### Test: Search + Tags + 3d Touch
1. Log into your account
2. Press over the SearchBar
3. Type the name of any of your tags

- [x] Verify that performing 3d Touch over any of the Tag rows does nothing
- [x] Verify that performing the horizontal swipe gesture, over a Tag Row, does nothing

### Test: Search is Global
1. Log into your account
2. Select an empty Tag
3. Press over the SearchBar
4. Type the content of any of your notes

- [x] Verify that the results are (global) and the relevant notes do show up. (Before this PR search was limited to the current tag)
- [x] Verify that pressing **Cancel** gets you back to the empty results UI

### Test: Transitions
1. Log into your account
2. Press over the Search Bar
3. Enter any keyword that yields results

- [x] Verify that pressing over any of the Notes causes the Editor to get pushed, without animations

### Test: Simperium Sync + Notes
1. Log into two devices, in the same account (Web and iOS works!)
2. Edit any of the notes that show up in iOS, elsewhere

- [x] Verify that iOS immediately updates the Notes List

### Test: Simperium Sync + Tags
1. Log into two devices, in the same account (Web and iOS works!)
2. Press over the Search Bar
3. Enter any keyword that yields Tag Results
4. Edit any of the Tag Names that do show up, elsewhere

- [x] Verify that the Tags do get immediately refreshed, even when in Search Mode!

### Test: Empty State
1. Log into your account
2. Open a tag that has no notes

- [x] Verify that Tags with no notes render the **No Notes** legend

### Test: No Results
1. Log into your account
2. Press over the Search Bar
3. Type a keyword that yields no results

- [x] Verify that Tags with no notes render the **No Results** legend

### Release
These changes do not require release notes.
